kernel/src/mem.rs

```diff
@@ -91,7 +91,10 @@ impl<L, T> AppSlice<L, T> {
                 .process
                 .kernel
                 .process_map_or(false, appid.idx(), |process| {
-                    process.add_mpu_region(self.ptr() as *const u8, self.len() as u32)
+                    match process.add_mpu_region(self.ptr() as *const u8, self.len(), self.len()) {
+                        Some(_) => true,
+                        None => false,
+                    }
```

Comment:
I think this can just be `process.add_mpu_region(self.ptr() as *const u8, self.len(), self.len()).is_ok()`, but I'm not completely sure.